src/main/java/com/palantir/gradle/gitversion/Git.java

```diff
@@ -17,130 +17,65 @@
 package com.palantir.gradle.gitversion;
 
 import java.io.File;
-import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
+import org.gradle.api.provider.Provider;
 import org.gradle.api.provider.ProviderFactory;
 import org.gradle.process.ExecOutput;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+/**
+ * Runs git commands in a gradle-aware manner, so gradle can do up-to-date checking for the configuration cache.
+ */
 class Git {
     private static final Logger log = LoggerFactory.getLogger(Git.class);
 
-    private final File directory;
+    private final File repositoryDir;
     private final ProviderFactory providerFactory;
 
-    Git(File directory, ProviderFactory providerFactory) {
+    Git(File repositoryDir, ProviderFactory providerFactory) {
         this.providerFactory = providerFactory;
-        this.directory = directory;
+        this.repositoryDir = repositoryDir;
         if (!gitCommandExists()) {
-            throw new RuntimeException("Git not found in project");
+            throw new RuntimeException("git command not found in project.");
         }
     }
 
-    private String runGitCmd(String... commands) throws IOException, InterruptedException {
-        return runGitCmd(new HashMap<>(), commands);
+    private boolean gitCommandExists() {
+        return providerFactory
```

Comment:
I still don't believe this should be here. It's not even tied to this particular git repo.

You also want to ensure with `execSpec.setIgnoreExitValue(true)` otherwise you're not actually going to be able to throw the exception providing a nice helper i.e. `"git command not found in project"`.

Reply:
Yep i see your point now. If git didn't exist, it should tell us in the error message when we try to use git anyways.